src/zenml/integrations/kubeflow/orchestrators/kubeflow_component.py

```diff
@@ -109,8 +108,23 @@ def __init__(
           tfx_ir: The TFX intermedia representation of the pipeline.
           pod_labels_to_attach: Dict of pod labels to attach to the GKE pod.
           runtime_parameters: Runtime parameters of the pipeline.
-          metadata_ui_path: File location for metadata-ui-metadata.json file.
         """
+        # Path to a metadata file that will be displayed in the KFP UI
+        # This metadata file needs to be in a mounted emptyDir to avoid
+        # sporadic failures with the (not mature) PNS executor
+        # See these links for more information about limitations of PNS +
+        # security context:
+        # https://www.kubeflow.org/docs/components/pipelines/installation/localcluster-deployment/#deploying-kubeflow-pipelines
+        # https://argoproj.github.io/argo-workflows/empty-dir/
+        # KFP will switch to the Emissary executor (soon), when this emptyDir
+        # mount will not be necessary anymore, but for now it's still in alpha
+        # status (https://www.kubeflow.org/docs/components/pipelines/installation/choose-executor/#emissary-executor)
+        metadata_ui_path = "/outputs/mlpipeline-ui-metadata.json"
```

Comment:
I had to change this from `/mlpipeline-ui-metadata.json` to `/tmp/mlpipeline-ui-metadata.json` recently because the local user didn't have permissions to write to the root directory. Did you check that this file is actually created by the pipeline steps ? You should see something in the step visualizations in the Kubeflow UI, if it works.

Reply:
Works for me, it shows the Markdown with all the execution parameters/inputs/outputs as usual

Reply:
Great !